CallKitDemo/OTDefaultAudioDevice.m

```diff
@@ -633,11 +633,16 @@ - (void) handleInterruptionEvent:(NSInteger) interruptionType
                                                [self handleInterruptionEvent:
                                                 AVAudioSessionInterruptionTypeEnded];
                                            });
-                        } else
+                        } else // This shouldn't happen!
                         {
-                            // This shouldn't happen!
-                            isRecorderInterrupted = NO;
-                            isPlayerInterrupted = NO;
+                            // When a call get resumed, we get first audio interruption notification from iOS
+                            // but the audio fails, since iOS holds the audio session until user presses "Unhold call",
+                            // When user press "Unhold call" we post a fake interruption notification from "didActivate" audioSession callback
```

Comment:
Is it neccesary to manually click on `unhold call`, can't we do it automatically in the `applicationDidBecomeActive(_:)` callback?

Reply:
Yes, It is necessary and it is CallKit default behavior, we can fix that by automatically "unhold the call" when appDidBecomeActive but think about this use case : 
If "User A" on OpenTok Call and is "On Hold", also he is on regular phone call with "User B". if "User B" somehow disconnect from his end, it is to prevent "User A", not to continue his talk with "OpenTok" by assuming he is on still regular phone call.

If User A disconnects the regular phone call on his end, the CallKit automatically call "OnHold" callback with "false" and we automatically resume OpenTok but It is not the case if the remote user does that as I explained above.

WhatsApp and other VOIP apps works the same way!